engine/enginedeck: Don't stop playback if no passthru input is configured

Holzhaus opened the pull request:
This is related to PR #2466 and this discussion on Zulip:
https://mixxx.zulipchat.com/#narrow/stream/109122-general/topic/PASS.20button

Holzhaus commented:
Pinging @ronso0 for review ;-)

ronso0 commented:
woooh, was it that easy? nice

ywwg commented:
oh sweet thanks

daschuer commented:
The ci fails are unrelated

daschuer commented on src/engine/enginebuffer.cpp:
```diff
@@ -575,6 +576,12 @@ void EngineBuffer::ejectTrack() {
 
 void EngineBuffer::slotPassthroughChanged(double enabled) {
     if (enabled) {
+        // If no vinyl input is configured, do nothing
```
We have here the issue that "passthrough" is used in other places to check if it is active. 
When ignoring it here, we get an ambiguous states. 
I see two possible solutions:
* Don't rely on "pathtrough" eselwhere.
* Use connect valueChangeRequest and reject passthrough if it is not configured.

ywwg replied:
> Use connect valueChangeRequest and reject passthrough if it is not configured.

this feels right to me -- rejecting the change would cause the pass button to turn off, correctly. the user may wonder why passthrough isn't turning on, but that's more likely to send them to the sound preferences than it turning on and just not working

Holzhaus commented:
This now uses `connectValueChangeRequest` and still shows a popup in case you click passthrough with no input configured.

daschuer commented on src/mixer/basetrackplayer.cpp:
```diff
@@ -41,6 +41,8 @@ BaseTrackPlayerImpl::BaseTrackPlayerImpl(QObject* pParent,
     m_pChannel = new EngineDeck(channelGroup, pConfig, pMixingEngine,
                                 pEffectsManager, defaultOrientation);
+    connect(m_pChannel, &EngineDeck::noPassthroughInputConfigured,
+            [this]{ emit(noPassthroughInputConfigured()); });
```
clazy wants emit without ()

Holzhaus replied:
oops, fixed it.

Holzhaus committed:
engine/enginedeck: Ensure that configured input before enabing passthru

This avoids stopping the playing deck if no input is configured for
passthrough.

Holzhaus committed:
engine/enginedeck: Reenable popup when activating unconfigured passthru

daschuer commented:
BaseTrackPlayerImpl::slotPassthroughEnabled
https://github.com/mixxxdj/mixxx/blob/30674ab04d1c8de320792a7dd6794db5eec423a9/src/mixer/basetrackplayer.cpp#L413
can be disposed now.

daschuer commented:
You can connect noDeckPassthroughInputConfigured directly using pDeck->getEngineDeck();

Does noVinylControlInputConfigured suffer from the same issues?

Holzhaus committed:
mixer/basetrackplayer: Remove obsolete passthrough enabled handling

Holzhaus commented:
> Does noVinylControlInputConfigured suffer from the same issues?

No, I don't think so.

Holzhaus committed:
engine/enginedeck: Fix clazy warning regarding emit parentheses

Holzhaus committed:
mixer: Connect noPassthroughInputConfigured directly via EngineDeck

ywwg commented:
> No, I don't think so.

the prefs window should pop up if the user tries to enable vinyl control with no input selected

Holzhaus commented:
>>No, I don't think so.
>
>the prefs window should pop up if the user tries to enable vinyl control with no input selected

Doesn't it? I thought it did when I tested it.

ywwg commented:
> Doesn't it? I thought it did when I tested it.

? Are you saying it does suffer the same issue, or it does pop up the prefs window?  It *should* pop up the prefs window.

Holzhaus commented:
> > Doesn't it? I thought it did when I tested it.
> 
> ? Are you saying it does suffer the same issue, or it does pop up the prefs window? It _should_ pop up the prefs window.

It does pop up the window, both with and without these changes. But it doesn't stop playback when activated while the input is unconfigured (which is the issue I thought we were talking about).

ywwg commented:
ah, ok, sorry.  right, correct, it doesn't stop playback

daschuer commented:
LGTM, Thank you.
